packages/api/cms-api/src/content-generation/openai-content-generation.service.ts

```diff
@@ -0,0 +1,78 @@
+import { AzureKeyCredential, ChatRequestMessage, OpenAIClient } from "@azure/openai";
+import { Injectable } from "@nestjs/common";
+import fetch from "node-fetch";
+
+import { OpenAiContentGenerationConfig } from "./content-generation.config";
+import { ContentGenerationServiceInterface } from "./content-generation.service.interface";
+
+@Injectable()
+export class OpenAiContentGenerationService implements ContentGenerationServiceInterface {
+    constructor(private readonly config: OpenAiContentGenerationConfig<ContentGenerationServiceInterface>) {}
+
+    async generateAltText(fileUrl: string): Promise<string> {
+        if (!this.config.generateAltText) {
+            return "Model not defined";
+        }
+
+        const client = new OpenAIClient(this.config.generateAltText.apiUrl, new AzureKeyCredential(this.config.generateAltText.apiKey));
+        const prompt: ChatRequestMessage[] = [
+            {
+                role: "system",
+                content:
+                    "You are an alt text expert. The user will provide you with an image and your job is to provide a description of the image. This will help visually impaired users understand the content of the image. Keep yourself to a short description",
+            },
+            {
+                role: "user",
+                content: [
+                    {
+                        type: "image_url",
+                        imageUrl: {
+                            url: await this.loadImageAsBase64(fileUrl),
```

Comment:
Now I see, you convert it to base64 here. So it will work. But I don't like that the base64 conversion is done here. 

If you make a custom implementation of this service, you have to know that the library will give you a private URL that you can't send to the model directly. To make it worse: If you don't know that, the problem won't occur locally, but only once deployed.

The service should get input that it can work with directly, e.g.

- a public fileUrl
- a base64 string
- the file ID (my preference because it's so flexible)